libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -301,39 +321,25 @@ func (p *pod) emitEvents(pod *kubernetes.Pod, flag string, containers []kubernet
 		// Information that can be used in discovering a workload
 		kubemeta := meta.Clone()
 		kubemeta["container"] = cmeta
-
-		// Pass annotations to all events so that it can be used in templating and by annotation builders.
-		annotations := common.MapStr{}
-		for k, v := range pod.GetObjectMeta().GetAnnotations() {
-			safemapstr.Put(annotations, k, v)
-		}
 		kubemeta["annotations"] = annotations
-		if p.namespaceWatcher != nil {
-			if rawNs, ok, err := p.namespaceWatcher.Store().GetByKey(pod.Namespace); ok && err == nil {
-				if namespace, ok := rawNs.(*kubernetes.Namespace); ok {
-					nsAnn := common.MapStr{}
-
-					for k, v := range namespace.GetAnnotations() {
-						safemapstr.Put(nsAnn, k, v)
-					}
-					kubemeta["namespace_annotations"] = nsAnn
-				}
-			}
-		}
+		kubemeta["namespace_annotations"] = nsAnn
 
 		// Without this check there would be overlapping configurations with and without ports.
 		if len(c.Ports) == 0 {
+			// Set a zero port on the event to signify that the event is from a container
```

Comment:
Should we better use an explicit field to indicate what kind of resource originated the event?

Reply:
im open to suggestions?

Reply:
:+1 on having sth like `source: pod/container` having defined as consts the possible types.

Reply:
Maybe we can use `resource`, so this indicates what resource the event is representing.

Reply:
We talked a little bit about this offline, the point in favour of keeping this event as is is that so far these events are resource-agnostic. After that I think that I agree that we can consider these events as "events with a port, but port is not set". Let's continue with this.